drivers/platform/xilinx/uart.c

```diff
@@ -149,17 +149,19 @@ int32_t uart_read(struct uart_desc *desc, uint8_t *data, uint32_t bytes_number)
 int32_t uart_write(struct uart_desc *desc, const uint8_t *data,
 		   uint32_t bytes_number)
 {
+#define WRITE_SIZE 255
 	struct xil_uart_desc *xil_uart_desc = desc->extra;
-	size_t total_sent;
+	size_t bytes_sent, offset = 0, len;
 
 	switch(xil_uart_desc->type) {
 	case UART_PS:
 #ifdef XUARTPS_H
-		total_sent = XUartPs_Send(xil_uart_desc->instance, (u8*)data,
-					  bytes_number);
-		while (XUartPs_IsSending(xil_uart_desc->instance));
-		if (total_sent < bytes_number)
-			return FAILURE;
+		while (offset < bytes_number) {
+			len = offset + WRITE_SIZE > bytes_number ? (bytes_number - offset) : WRITE_SIZE;
```

Comment:
I think is more understandable if you write it this way
`len =  ((bytes_number - offset) < WRITE_SIZE) ? (bytes_number - offset) : WRITE_SIZE;`

Reply:
Good suggestion! Done.